plugins/cookie_manager/lib/src/cookie_mgr.dart

```diff
@@ -29,6 +29,17 @@ class CookieManager extends Interceptor {
 
   /// Merge cookies into a Cookie string.
   static String getCookies(List<Cookie> cookies) {
+    // remove previous cookie with same name
+    cookies = cookies.fold<List<Cookie>>(
+        [],
+        (prev, current) => prev.indexWhere((c) => c.name == current.name) == -1
+            ? [...prev, current]
+            : prev);
```

Comment:
Can you use control flows rather than inline APIs, which will make the code more readable and reliable?